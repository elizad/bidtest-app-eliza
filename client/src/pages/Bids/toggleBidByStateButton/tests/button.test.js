import React from 'react'
import ReactDOM from 'react-dom'
import ToggleBidByStateButton from '../toggleBidByStateButton.js'

import { render, cleanup } from '@testing-library/react'


import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<ToggleBidByStateButton />, div)
});

it('renders button correctly', () => {
  const { getByTestId } = render(<ToggleBidByStateButton label="click me please" />)
  expect(getByTestId('button')).toHaveTextContent('Show All/Accepted')
});

it('renders button correctly', () => {
  const { getByTestId } = render(<ToggleBidByStateButton label="save" />)
  expect(getByTestId('button')).toHaveTextContent('Show All/Accepted')
});


it('matches snapshot 2', () => {
  const tree = renderer.create(<ToggleBidByStateButton label="click me please" />).toJSON()
  expect(tree).toMatchSnapshot()
});
