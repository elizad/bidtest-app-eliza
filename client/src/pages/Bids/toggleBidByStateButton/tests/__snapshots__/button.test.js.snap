// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 2 1`] = `
<button
  className="button-style btn"
  data-testid="button"
  name="showTrueAll"
>
  Show All/Accepted
</button>
`;
